src/rgw/rgw_iam_policy.cc

```diff
@@ -719,7 +719,7 @@ static optional<Principal> parse_principal(CephContext* cct, TokenID t,
 			  ECMAScript | optimize);
     smatch match;
     if (regex_match(a->resource, match, rx)) {
-      ceph_assert(match.size() == 2);
+      ceph_assert(match.size() == 3);
```

Comment:
If we have a Principal ARN of the form -> arn:aws:iam::usfolks:user/fred in the Bucket Policy, then match[0] will be the full string, match[1] will be 'user' and match[2] will be 'fred', hence match size will be 3.